File: Week_02/Seminar/TK/contains_duplicate/solution_set.cpp
#include <vector>
#include <unordered_set>

class Solution
{
public:
    bool containsDuplicate(std::vector<int> &nums)
    {
        std::unordered_set<int> s(nums.begin(), nums.end());
        return s.size() != nums.size();
    }
};

File: Week_08/Seminar/Solutions/contains_duplicate_ii/solution_set.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        if k == 0:
            return False
            
        s = set()
        left = 0

        for right in range(len(nums)):
            if nums[right] in s:
                return True

            if right >= k:
                s.remove(nums[left])
                left += 1

            s.add(nums[right])

        return False
    